Dockerfile

```diff
@@ -1,17 +1,15 @@
-FROM ruby:2.3-alpine
+FROM ruby:2.4-alpine
+
+RUN adduser -u 9000 -D app
 
 WORKDIR /usr/src/app
-COPY Gemfile /usr/src/app/
-COPY Gemfile.lock /usr/src/app/
 
-RUN gem install bundler && \
-    bundle install -j 4 && \
-    rm -fr /usr/share/ri
+COPY Gemfile Gemfile.lock ./
 
-RUN apk --update add git
+RUN gem install --no-document -g
```

Comment:
Reverted to Bundler since it's easier than finding any formal `gem i -g` endorsements.

However:

* The CVE is not relevant to this particular case. We have only one source.
* The CVE was partially addressed in Rubygems. The proposed bundler fix was implemented except for emitted warning.
* Anecdotal "for": apart from many of my own projects, we have successfully used `gem i -g` in the Grep engine.